File: ios/LaneShadow/Sandbox/Stories/Molecules/LSNavigatorMoleculesStory.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

/// Navigator molecule stories — Sprint 4 Navigator components
@MainActor
enum LSNavigatorMoleculesStories {
    static let all: [Story] = phaseIndicatorStories + weatherTimelineStories + instrumentReadoutStories

    // MARK: - LSPhaseIndicator Stories

    static var phaseIndicatorStories: [Story] {
        [
            Story(
                id: "molecules.phaseIndicator.default",
                tier: .molecule,
                component: "LSPhaseIndicator",
                name: "Phase Indicator - In Progress",
                summary: "Shows compass chip, narrator header, and multi-step phase list with active step animation"
            ) { _ in
                LSPhaseIndicator(
                    phases: [
                        PlanningPhase(id: "1", label: "Understanding your request", state: .done),
                        PlanningPhase(id: "2", label: "Searching routes", state: .active),
                        PlanningPhase(id: "3", label: "Checking conditions", state: .pending),
                        PlanningPhase(id: "4", label: "Evaluating options", state: .pending),
                    ],
                    header: "Let me think on that…"
                )
            },

            Story(
                id: "molecules.phaseIndicator.allDone",
                tier: .molecule,
                component: "LSPhaseIndicator",
                name: "Phase Indicator - All Done",
                summary: "Shows completed state with all steps done and green top border"
            ) { _ in
                LSPhaseIndicator(
                    phases: [
                        PlanningPhase(id: "1", label: "Understanding your request", state: .done),
                        PlanningPhase(id: "2", label: "Searching routes", state: .done),
                        PlanningPhase(id: "3", label: "Checking conditions", state: .done),
                        PlanningPhase(id: "4", label: "Evaluating options", state: .done),
                    ],
                    header: "Found 3 great routes"
                )
            },

            Story(
                id: "molecules.phaseIndicator.allPending",
                tier: .molecule,
                component: "LSPhaseIndicator",
                name: "Phase Indicator - All Pending",
                summary: "Shows initial state with all steps pending"
            ) { _ in
                LSPhaseIndicator(
                    phases: [
                        PlanningPhase(id: "1", label: "Understanding your request", state: .pending),
                        PlanningPhase(id: "2", label: "Searching routes", state: .pending),
                        PlanningPhase(id: "3", label: "Checking conditions", state: .pending),
                    ],
                    header: "Starting search…"
                )
            },
        ]
    }

    // MARK: - LSWeatherTimeline Stories

    static var weatherTimelineStories: [Story] {
        [
            Story(
                id: "molecules.weatherTimeline.sixHours",
                tier: .molecule,
                component: "LSWeatherTimeline",
                name: "Weather Timeline - 6 Hours",
                summary: "Shows 6-cell horizontal grid with per-condition tinted backgrounds"
            ) { _ in
                LSWeatherTimeline(
                    entries: [
                        WeatherEntry(hour: "9 AM", condition: .clear, temp: "68°"),
                        WeatherEntry(hour: "10 AM", condition: .clear, temp: "70°"),
                        WeatherEntry(hour: "11 AM", condition: .rain, temp: "65°"),
                        WeatherEntry(hour: "12 PM", condition: .rain, temp: "63°"),
                        WeatherEntry(hour: "1 PM", condition: .wind, temp: "64°"),
                        WeatherEntry(hour: "2 PM", condition: .clear, temp: "67°"),
                    ],
                    from: "9 AM",
                    to: "2 PM"
                )
            },

            Story(
                id: "molecules.weatherTimeline.mixedWeather",
                tier: .molecule,
                component: "LSWeatherTimeline",
                name: "Weather Timeline - Mixed Weather",
                summary: "Shows all six weather conditions with distinct tint colors"
            ) { _ in
                LSWeatherTimeline(
                    entries: [
                        WeatherEntry(hour: "9 AM", condition: .clear, temp: "68°"),
                        WeatherEntry(hour: "10 AM", condition: .rain, temp: "65°"),
                        WeatherEntry(hour: "11 AM", condition: .wind, temp: "63°"),
                        WeatherEntry(hour: "12 PM", condition: .storm, temp: "60°"),
                        WeatherEntry(hour: "1 PM", condition: .hot, temp: "75°"),
                        WeatherEntry(hour: "2 PM", condition: .cold, temp: "55°"),
                    ],
                    from: "9 AM",
                    to: "2 PM"
                )
            },

            Story(
                id: "molecules.weatherTimeline.allClear",
                tier: .molecule,
                component: "LSWeatherTimeline",
                name: "Weather Timeline - All Clear",
                summary: "Shows consistent clear conditions across all hours"
            ) { _ in
                LSWeatherTimeline(
                    entries: [
                        WeatherEntry(hour: "9 AM", condition: .clear, temp: "68°"),
                        WeatherEntry(hour: "10 AM", condition: .clear, temp: "70°"),
                        WeatherEntry(hour: "11 AM", condition: .clear, temp: "72°"),
                        WeatherEntry(hour: "12 PM", condition: .clear, temp: "74°"),
                        WeatherEntry(hour: "1 PM", condition: .clear, temp: "75°"),
                        WeatherEntry(hour: "2 PM", condition: .clear, temp: "73°"),
                    ],
                    from: "9 AM",
                    to: "2 PM"
                )
            },
        ]
    }

    // MARK: - LSInstrumentReadout Stories

    static var instrumentReadoutStories: [Story] {
        [
            Story(
                id: "molecules.instrumentReadout.fourMetrics",
                tier: .molecule,
                component: "LSInstrumentReadout",
                name: "Instrument Readout - 4 Metrics",
                summary: "Shows 4-column grid with distance, time, climb, and scenic score"
            ) { _ in
                LSInstrumentReadout(
                    metrics: [
                        .dist("64 mi"),
                        .time("2h 10m"),
                        .climb("2,400ft"),
                        .scenic("9.2"),
                    ]
                )
            },

            Story(
                id: "molecules.instrumentReadout.threeMetrics",
                tier: .molecule,
                component: "LSInstrumentReadout",
                name: "Instrument Readout - 3 Metrics",
                summary: "Shows 3-column grid with distance, time, and climb"
            ) { _ in
                LSInstrumentReadout(
                    metrics: [
                        .dist("64 mi"),
                        .time("2h 10m"),
                        .climb("2,400ft"),
                    ]
                )
            },

            Story(
                id: "molecules.instrumentReadout.longValues",
                tier: .molecule,
                component: "LSInstrumentReadout",
                name: "Instrument Readout - Long Values",
                summary: "Shows metric grid with longer values and units"
            ) { _ in
                LSInstrumentReadout(
                    metrics: [
                        InstrumentMetric(label: "Distance", value: "164.2", unit: "miles"),
                        InstrumentMetric(label: "Duration", value: "12h 45m"),
                        InstrumentMetric(label: "Elevation", value: "12,400", unit: "ft gain"),
                        InstrumentMetric(label: "Scenic", value: "9.8", isAccent: true),
                    ]
                )
            },
        ]
    }
}

File: android/app/src/debug/java/com/laneshadow/sandbox/stories/molecules/LSNavigatorMoleculesStory.kt
package com.laneshadow.sandbox.stories.molecules

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.laneshadow.theme.LaneShadowTheme
import com.laneshadow.ui.atoms.PhaseDotState
import com.laneshadow.ui.molecules.InstrumentMetric
import com.laneshadow.ui.molecules.LSInstrumentReadout
import com.laneshadow.ui.molecules.LSPhaseIndicator
import com.laneshadow.ui.molecules.LSWeatherTimeline
import com.laneshadow.ui.molecules.PlanningPhase
import com.laneshadow.ui.molecules.WeatherCondition
import com.laneshadow.ui.molecules.WeatherTimelineEntry
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSNavigatorMoleculesStory {
    val all: List<Story> = listOf(
        // LSPhaseIndicator stories
        Story(
            id = "molecules.phase.default",
            tier = ComponentTier.Molecule,
            component = "LSPhaseIndicator",
            name = "Default",
            summary = "Phase indicator with one active step.",
            content = { PhaseIndicatorDefaultStory() },
        ),
        Story(
            id = "molecules.phase.all-done",
            tier = ComponentTier.Molecule,
            component = "LSPhaseIndicator",
            name = "All Done",
            summary = "Phase indicator with all steps complete.",
            content = { PhaseIndicatorAllDoneStory() },
        ),
        Story(
            id = "molecules.phase.all-pending",
            tier = ComponentTier.Molecule,
            component = "LSPhaseIndicator",
            name = "All Pending",
            summary = "Phase indicator with all steps pending.",
            content = { PhaseIndicatorAllPendingStory() },
        ),

        // LSWeatherTimeline stories
        Story(
            id = "molecules.weather.timeline.6-hours",
            tier = ComponentTier.Molecule,
            component = "LSWeatherTimeline",
            name = "6 Hours",
            summary = "Weather timeline with 6 hourly cells.",
            content = { WeatherTimeline6HoursStory() },
        ),
        Story(
            id = "molecules.weather.timeline.mixed",
            tier = ComponentTier.Molecule,
            component = "LSWeatherTimeline",
            name = "Mixed Weather",
            summary = "Weather timeline with mixed conditions.",
            content = { WeatherTimelineMixedStory() },
        ),
        Story(
            id = "molecules.weather.timeline.all-clear",
            tier = ComponentTier.Molecule,
            component = "LSWeatherTimeline",
            name = "All Clear",
            summary = "Weather timeline with all clear conditions.",
            content = { WeatherTimelineAllClearStory() },
        ),

        // LSInstrumentReadout stories
        Story(
            id = "molecules.instrument.4-metrics",
            tier = ComponentTier.Molecule,
            component = "LSInstrumentReadout",
            name = "4 Metrics",
            summary = "Instrument readout with 4 metrics.",
            content = { InstrumentReadout4MetricsStory() },
        ),
        Story(
            id = "molecules.instrument.3-metrics",
            tier = ComponentTier.Molecule,
            component = "LSInstrumentReadout",
            name = "3 Metrics",
            summary = "Instrument readout with 3 metrics.",
            content = { InstrumentReadout3MetricsStory() },
        ),
        Story(
            id = "molecules.instrument.long-values",
            tier = ComponentTier.Molecule,
            component = "LSInstrumentReadout",
            name = "Long Values",
            summary = "Instrument readout with long metric values.",
            content = { InstrumentReadoutLongValuesStory() },
        ),
    )
}

// LSPhaseIndicator stories

@Composable
private fun PhaseIndicatorDefaultStory() {
    LaneShadowTheme {
        StoryColumn {
            LSPhaseIndicator(
                phases = listOf(
                    PlanningPhase("Understanding your request", PhaseDotState.Done),
                    PlanningPhase("Searching routes", PhaseDotState.Active),
                    PlanningPhase("Checking conditions", PhaseDotState.Pending),
                    PlanningPhase("Evaluating options", PhaseDotState.Pending),
                    PlanningPhase("Finalizing plan", PhaseDotState.Pending),
                ),
                header = "Let me think on that…",
            )
        }
    }
}

@Composable
private fun PhaseIndicatorAllDoneStory() {
    LaneShadowTheme {
        StoryColumn {
            LSPhaseIndicator(
                phases = listOf(
                    PlanningPhase("Understanding your request", PhaseDotState.Done),
                    PlanningPhase("Searching routes", PhaseDotState.Done),
                    PlanningPhase("Checking conditions", PhaseDotState.Done),
                    PlanningPhase("Evaluating options", PhaseDotState.Done),
                    PlanningPhase("Finalizing plan", PhaseDotState.Done),
                ),
                header = "Route found!",
            )
        }
    }
}

@Composable
private fun PhaseIndicatorAllPendingStory() {
    LaneShadowTheme {
        StoryColumn {
            LSPhaseIndicator(
                phases = listOf(
                    PlanningPhase("Understanding your request", PhaseDotState.Pending),
                    PlanningPhase("Searching routes", PhaseDotState.Pending),
                    PlanningPhase("Checking conditions", PhaseDotState.Pending),
                    PlanningPhase("Evaluating options", PhaseDotState.Pending),
                    PlanningPhase("Finalizing plan", PhaseDotState.Pending),
                ),
                header = "Starting search…",
            )
        }
    }
}

// LSWeatherTimeline stories

@Composable
private fun WeatherTimeline6HoursStory() {
    LaneShadowTheme {
        StoryColumn {
            LSWeatherTimeline(
                entries = listOf(
                    WeatherTimelineEntry("9 AM", WeatherCondition.Clear, "68°"),
                    WeatherTimelineEntry("10 AM", WeatherCondition.Clear, "72°"),
                    WeatherTimelineEntry("11 AM", WeatherCondition.Clear, "75°"),
                    WeatherTimelineEntry("12 PM", WeatherCondition.Clear, "78°"),
                    WeatherTimelineEntry("1 PM", WeatherCondition.Clear, "76°"),
                    WeatherTimelineEntry("2 PM", WeatherCondition.Clear, "74°"),
                ),
                from = "9 AM",
                to = "2 PM",
            )
        }
    }
}

@Composable
private fun WeatherTimelineMixedStory() {
    LaneShadowTheme {
        StoryColumn {
            LSWeatherTimeline(
                entries = listOf(
                    WeatherTimelineEntry("9 AM", WeatherCondition.Clear, "68°"),
                    WeatherTimelineEntry("10 AM", WeatherCondition.Rain, "65°"),
                    WeatherTimelineEntry("11 AM", WeatherCondition.Rain, "63°"),
                    WeatherTimelineEntry("12 PM", WeatherCondition.Wind, "62°"),
                    WeatherTimelineEntry("1 PM", WeatherCondition.Clear, "70°"),
                    WeatherTimelineEntry("2 PM", WeatherCondition.Clear, "74°"),
                ),
                from = "9 AM",
                to = "2 PM",
            )
        }
    }
}

@Composable
private fun WeatherTimelineAllClearStory() {
    LaneShadowTheme {
        StoryColumn {
            LSWeatherTimeline(
                entries = listOf(
                    WeatherTimelineEntry("9 AM", WeatherCondition.Clear, "68°"),
                    WeatherTimelineEntry("10 AM", WeatherCondition.Clear, "72°"),
                    WeatherTimelineEntry("11 AM", WeatherCondition.Clear, "75°"),
                    WeatherTimelineEntry("12 PM", WeatherCondition.Clear, "78°"),
                    WeatherTimelineEntry("1 PM", WeatherCondition.Clear, "76°"),
                    WeatherTimelineEntry("2 PM", WeatherCondition.Clear, "74°"),
                    WeatherTimelineEntry("3 PM", WeatherCondition.Clear, "72°"),
                    WeatherTimelineEntry("4 PM", WeatherCondition.Clear, "70°"),
                ),
                from = "9 AM",
                to = "4 PM",
            )
        }
    }
}

// LSInstrumentReadout stories

@Composable
private fun InstrumentReadout4MetricsStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInstrumentReadout(
                metrics = listOf(
                    InstrumentMetric("Dist", "64 mi"),
                    InstrumentMetric("Time", "2h 10m"),
                    InstrumentMetric("Climb", "2,400ft"),
                    InstrumentMetric("Scenic", "9.2", isAccent = true),
                ),
            )
        }
    }
}

@Composable
private fun InstrumentReadout3MetricsStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInstrumentReadout(
                metrics = listOf(
                    InstrumentMetric("Dist", "42 mi"),
                    InstrumentMetric("Time", "1h 45m"),
                    InstrumentMetric("Climb", "1,800ft"),
                ),
            )
        }
    }
}

@Composable
private fun InstrumentReadoutLongValuesStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInstrumentReadout(
                metrics = listOf(
                    InstrumentMetric("Distance", "124.5 miles"),
                    InstrumentMetric("Duration", "5h 32m"),
                    InstrumentMetric("Elevation", "8,450ft gain"),
                    InstrumentMetric("Scenic", "9.7", isAccent = true),
                ),
            )
        }
    }
}

@Composable
private fun StoryColumn(content: @Composable () -> Unit) {
    Column(
        modifier = Modifier.padding(com.laneshadow.theme.LocalLaneShadowTheme.current.space.lg),
        verticalArrangement = Arrangement.spacedBy(com.laneshadow.theme.LocalLaneShadowTheme.current.space.md),
    ) {
        content()
    }
}
